bodhi/tests/server/consumers/test_masher.py

```diff
@@ -554,6 +563,36 @@ def test_testing_digest(self, mail, *args):
              '------------------------------------------------------------\n\n') % (
                 config.get('fedora_test_announce_list'), time.strftime('%Y'))), repr(body)
 
+    @mock.patch('bodhi.server.consumers.masher.ComposerThread.save_state')
+    def test_mash_invalid_dir(self, save_state):
+        msg = self._make_msg()
+        t = RPMComposerThread(msg['body']['msg']['composes'][0],
+                              'ralph', log, self.db_factory, self.tempdir)
+        t.devnull = mock.MagicMock()
+        t.id = 'f17-updates-testing'
+        with self.db_factory() as session:
+            t.db = session
+            t.compose = Compose.from_dict(session, msg['body']['msg']['composes'][0])
+            t.release = session.query(Release).filter_by(name=u'F17').one()
+            try:
+                fake_popen = mock.MagicMock()
+                fake_stdout = '''Some output
+Some more output ...... This is not a Compose dir: ....
+Compose dir: /tmp/nonsensical_directory
+That was the actual one'''
+                fake_popen.communicate = lambda: (fake_stdout, 'hello')
+                fake_popen.poll.return_value = None
+                fake_popen.returncode = 0
+                t._startyear = datetime.datetime.utcnow().year
+                t._wait_for_pungi(fake_popen)
+                assert False, "Mash with invalid directory did not crash"
+            except Exception as ex:
+                expected_error = ('Directory at /tmp/nonsensical_directory does not look like a '
+                                  'compose')
+                expected_error = expected_error.format(datetime.datetime.utcnow().year)
+                assert str(ex) == expected_error
+            t.db = None
```

Comment:
There's no problem leaving this line here, but you shouldn't need it because ```t``` will go out of scope once this test is done.